builtin/providers/aws/data_source_availability_zones.go

```diff
@@ -0,0 +1,50 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsAvailabilityZones() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsAvailabilityZonesRead,
+
+		Schema: map[string]*schema.Schema{
+			"available": &schema.Schema{
+				Type:     schema.TypeSet,
```

Comment:
I think some folks were interested in having these in a predictable order so that they can do stuff like pluck the first item from the list and use it as a "primary AZ" in configuration... so might be better to make this a `TypeList` and sort it alphabetically in the implementation so that the ordering is dependable.